Add CI

edgarrmondragon opened the pull request:
* Uses https://github.com/hynek/build-and-inspect-python-package to build the package
* Uses https://github.com/pypa/gh-action-pypi-publish to publish to PyPI
* Tests Python versions 3.8 - 3.13

Closes #4

barneygale commented:
I've added a trusted publisher on PyPI:

![image](https://github.com/barneygale/pathlib_abc/assets/960340/078c6f58-d0d9-4029-917b-0d6386cf1dd5)

Is there a box I need to tick in GitHub to enable CI to run on this PR?

edgarrmondragon commented:
> Thanks v much!
> 
> I've added a trusted publisher on PyPI:
> 
> ![image](https://private-user-images.githubusercontent.com/960340/294011332-078c6f58-d0d9-4029-917b-0d6386cf1dd5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQzMjgxMzYsIm5iZiI6MTcwNDMyNzgzNiwicGF0aCI6Ii85NjAzNDAvMjk0MDExMzMyLTA3OGM2ZjU4LWQwZDktNDAyOS05MTdiLTBkNjM4NmNmMWRkNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwNFQwMDIzNTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZjEyMTBjNzNlOTBiN2NhZTU2Y2JiZWZjZmIyNzk1MGFhYTkxYTUxNjQ4MTg4M2I1ZWI1NWIzMTQ0ODI2YTExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ky7vq4yoB2ld8G8qFCAxXfnLqpZnExv3xRaUrCynJGs)
> 
> Is there a box I need to tick in GitHub to enable CI to run on this PR?

@barneygale Unfortunately I think new workflows have to be merged into the default branch before they can run in PRs or be triggered manually. They've run in my fork if you wanna take a look:

* Build: https://github.com/edgarrmondragon/pathlib_abc/actions/runs/7394303399
* Test: https://github.com/edgarrmondragon/pathlib_abc/actions/runs/7394303398

---

I've updated the build workflow to use `release` environment, so we should probably make sure it exists: 
<img width="1120" alt="Screenshot 2024-01-03 at 7 34 11 p m" src="https://github.com/barneygale/pathlib_abc/assets/16805946/96217843-8bf3-4686-8ebc-1dff72a9f568">

The environment name 👆 should name `release` for this repo.

---

PS: Thanks for the invitation to collaborate!

barneygale commented:
Thanks again :-)
